### PR TITLE
[TORCH] Add aten.std e2e support

### PR DESCRIPTION
**Pull request by cathyzhyi**

### CODE REVIEW

**Comment by cathyzhyi** [comment]
This is needed for testing in https://github.com/llvm/torch-mlir/pull/538

**Review by pashu123 — APPROVED** [review]
Some nits.

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 1]
```diff
@@ -489,6 +489,92 @@ class DecomposeAtenMeanOp : public OpRewritePattern<AtenMeanOp> {
...
+                                PatternRewriter &rewriter) const override {
+    Location loc = op.getLoc();
+    Value self = op.self();
+    BaseTensorType inputTensorTy = self.getType().cast<BaseTensorType>();
+    if (!inputTensorTy.hasDtype() ||
+        !inputTensorTy.getDtype().isa<mlir::FloatType>()) {
+      return rewriter.notifyMatchFailure(op,
+                                         "Only aten.var support floating type");
+    }
+    Type rank0FloatTensorTy = op.getType();
```
[the remark this answers is not in the document]
Comment (cathyzhyi) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:521]: Yea, it's safer to add an assert here to make sure the type is as expected. Adding this.

File: lib/Dialect/Torch/Transforms/RefineTypes.cpp [hunk 2]
```diff
@@ -457,6 +457,9 @@ class TypeAnalyzer : public ForwardDataFlowAnalysis<ValueKnowledge> {
       Type dtype =
           getDtypeOrDefault(mean.getContext(), mean.dtype(), defaultDtype);
       return visitReductionAlongAllDimsOp(mean, dtype, operands);
+    } else if (isa<AtenStdOp, AtenVarOp>(op)) {
+      auto input = operands[0]->getValue();
```
[the remark this answers is not in the document]
Comment (cathyzhyi) [reply lib/Dialect/Torch/Transforms/RefineTypes.cpp:461]: oh, actually I'd prefer to assign to a variable called `input` first so it's more clear the dtype comes from `input`

File: lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp [hunk 3]
```diff
@@ -730,6 +818,12 @@ class DecomposeComplexOpsPass
     target.addIllegalOp<AtenArangeStartOp>();
     patterns.add<DecomposeAtenArgMaxOp>(context);
     target.addIllegalOp<AtenArgmaxOp>();
+    patterns.add<DecomposeAtenSquareOp>(context);
+    target.addIllegalOp<AtenSquareOp>();
+    patterns.add<DecomposeAtenVarOp>(context);
+    target.addIllegalOp<AtenVarOp>();
+    patterns.add<DecomposeAtenStdOp>(context);
+    target.addIllegalOp<AtenStdOp>();
```
Comment (joker-eph) [inline lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:826]: Just driving by and wondering: there aren't any FileCheck tests for the C++ code?

Comment (cathyzhyi) [reply lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp:826]: yea... we didn't enforce that. Usually just add e2e test. I should add that as a good practice.

**Pull request merged**